homeassistant/components/reolink/entity.py

```diff
@@ -128,3 +140,23 @@ def __init__(
                 sw_version=self._host.api.camera_sw_version(dev_ch),
                 configuration_url=self._conf_url,
             )
+
+    async def async_added_to_hass(self) -> None:
+        """Entity created."""
+        await super().async_added_to_hass()
+        cmd_key = self.entity_description.cmd_key
+        if cmd_key is not None:
+            self._update_cmd_list_count[cmd_key].setdefault(self._channel, 0)
+            self._update_cmd_list_count[cmd_key][self._channel] += 1
+            if self._channel not in self._host.update_cmd_list[cmd_key]:
+                self._host.update_cmd_list[cmd_key].append(self._channel)
```

Comment:
If you are going to do ref counting and channel is hashable it might make this a bit simpler to make `update_cmd_list` a `defaultdict` of `dict` and do

`self._host.update_cmd[cmd_key][self._channel] += 1` and `self._host.update_cmd[cmd_key][self._channel] -= 1`

Reply:
Thank you for the suggestion, I have implemented that now (I think).

Reply:
I now have two lists (update_cmd_list and update_cmd_list_count), I do that because I want to keep update_cmd_list as simple as possible since it is looked up many times for each update that happens every 1 minute.

Reply:
Where does the lookup happen?

Reply:
In the upstream reolink-aio library:
https://github.com/starkillerOG/reolink_aio/blob/714e87f05669c1b0889ba5c0718297ce6a06a10c/reolink_aio/api.py#L1533-L1537

For each command a check is made if it needs to be included for that channel:
https://github.com/starkillerOG/reolink_aio/blob/714e87f05669c1b0889ba5c0718297ce6a06a10c/reolink_aio/api.py#L1558-L1598

Reply:
In the future you should be able to optimize https://github.com/starkillerOG/reolink_aio/blob/714e87f05669c1b0889ba5c0718297ce6a06a10c/reolink_aio/api.py#L1546 a bit more by having all the commands in a `set` and than using `intersection` on the `dict` to find the supported commands

Reply:
That is very instresting, will keep this in mind.
You are so good in runtime optimazations (I read the members discord and I am amazed by your work).